java/flight/src/main/java/org/apache/arrow/flight/FlightStream.java

```diff
@@ -117,8 +137,12 @@ public void close() throws Exception {
         .map(t -> ((AutoCloseable) t))
         .collect(Collectors.toList());
 
+    final List<FieldVector> dictionaryVectors = dictionaries.getDictionaryIds().stream()
```

Comment:
Might it be worth factoring out some sort of `closeDictionaries`? I've written something like this a couple times now.

Reply:
The caveat with doing it here is that you have to remember to keep the dictionaries alive if you want to use the data beyond the stream lifetime.

Reply:
don't you have to do that anyways?

Reply:
Yes, it's just unfortunate that it's hard to get a compile-time guarantee.